sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsMeta.scala

```diff
@@ -807,6 +808,15 @@ abstract class BaseExprMeta[INPUT <: Expression](
   }
 
   final override def tagSelfForGpu(): Unit = {
+    val isConstantFoldingOff =
+      SQLConf.get.optimizerExcludedRules
+        .contains("org.apache.spark.sql.catalyst.optimizer.ConstantFolding")
+    if (isConstantFoldingOff) {
```

Comment:
nit: This looks great, but why do we need to check the config at all? `wrapped.foldable` should be cheap to do and it not directly impacted by the `optimizerExcludedRules`.